### PR TITLE
CSS: Remove fallback values from var() calls

### PR DESCRIPTION
**Pull request by acicovic**
## Description
In this PR, we're removing fallback values from `var()` calls for the variables we're declaring ourselves in `variables.scss`. Doing this uncovered some issues like erroneous variable names, and uncovered the lack of `var()` support in `gap`. Any such issues found have been fixed.

## Motivation and context
We're still using some variables where it makes sense to keep fallback values. However, for the variables we're declaring ourselves, it is better to fix any issues that may come up rather than rely on fallbacks.

## How has this been tested?
- Automated tests pass.
- Manually verified that the are no visual differences with `develop`.

## Summary by CodeRabbit

- **Style Updates**
    - Streamlined styling consistency by updating gap, padding, margin, font sizes, and color properties across components to utilize CSS variables.
    - Simplified styling declarations by replacing specific pixel values and fallbacks with variable references for enhanced flexibility.
    - Introduced base units for grid units and font sizes to ensure uniformity and maintainability in styling.

### CODE REVIEW

File: src/content-helper/common/css/variables.scss [hunk 1]
```diff
@@ -1,6 +1,15 @@
 /** SASS variables **/
 $html-font-size: 16px; // Used in functions.scss.
 
+/**
+ * Base units for internal use in this file. To be used externally only if var()
+ * is unsupported by the calling CSS property (e.g gap).
```
Comment (vaurdan) [inline src/content-helper/common/css/variables.scss]: TIL!

**Review by vaurdan — APPROVED** [review]
Thanks for working on this!

File: src/content-helper/editor-sidebar/performance-stats/performance-stats.scss [hunk 2]
```diff
@@ -94,23 +94,23 @@
 		display: flex;
 		align-items: flex-start;
 		align-content: flex-start;
-		gap: to_rem(16px) var(--grid-unit-20, to_rem(16px));
+		gap: to_rem(16px) var(--grid-unit-20);
```
[the remark this answers is not in the document]
Comment (acicovic) [reply src/content-helper/editor-sidebar/performance-stats/performance-stats.scss:97]: In your suggestion, you're suggesting using `var()` to assign a value to `gap`. However, my research and testing indicates that `gap` doesn't work with `var()`. What's your feedback regarding this?

[the remark this answers is not in the document]
Comment (acicovic) [reply src/content-helper/editor-sidebar/performance-stats/performance-stats.scss:97]: I've tested with the **redacted** browser, and `var()` assignments didn't work for `gap`. However, according to you it should. Do you have any specific sources that explicitly state or demonstrate that `var()` works with `gap`?

[the remark this answers is not in the document]
Comment (acicovic) [reply src/content-helper/editor-sidebar/performance-stats/performance-stats.scss:97]: Does `var()` work well with shorthand properties?

[the remark this answers is not in the document]
Comment (acicovic) [reply src/content-helper/editor-sidebar/performance-stats/performance-stats.scss:97]: When using `var()` with `gap`, what's the entire set of values that need to be given?

**Comment by acicovic** [comment]
@coderabbitai resolve

**Comment by acicovic** [comment]
@vaurdan, with the latest commit:
- We are stopping the use of `base-*` units outside of `variables.scss`.
- We have reverted to using `var()` for `gap` again.
- We've commented out some `gap`s that seem to have no effect. They are to be removed if we don't notice any UI glitches.

The reason I had been led to believe that `var()` wasn't working for `gap` was due to:
1. Copilot saying it was not supported (last paragraph).
    image
2. When testing and changing the value to `rem`, `px` or anything else, it was working. In my mind, this further confirmed that Copilot was right.
3. I couldn't find any documentation that strongly affirmed that `var()` should work with `gap`.

However, With the [review from CodeRabbit](https://github.com/Parsely/wp-parsely/pull/2314#discussion_r1550058019) I became quite skeptical and started testing again. It turned out that the variable I was testing with was out of scope in the assignment I was modifying. This explained why it wasn't working, and why changing it with a non-variable value made it work.

All in all, this is a very good demonstration of how we can get fooled, thinking that something doesn't work when it actually does. 🙂

**Comment by acicovic** [comment]
@coderabbitai, why isn't this PR approved from you yet?

**Comment by acicovic** [comment]
@coderabbitai, can you approve the PR then? It's still in "Changes Requested" mode from your end.

**Review by vaurdan — APPROVED** [review]

**Pull request merged**